db/bigtable_eth1.go

```diff
@@ -2817,20 +2817,11 @@ func (bigtable *Bigtable) GetMetadataForAddress(address []byte) (*types.Eth1Addr
 	sort.Slice(ret.Balances, func(i, j int) bool {
 		priceI := decimal.New(0, 0)
 		priceJ := decimal.New(0, 0)
-		var err error
-
-		if string(ret.Balances[i].Metadata.Price) != "" {
-			priceI, err = decimal.NewFromString(string(ret.Balances[i].Metadata.Price))
-			if err != nil {
-				logger.WithError(err).Errorf("error parsing string price value, price: %s", ret.Balances[i].Metadata.Price)
-			}
+		if len(ret.Balances[i].Metadata.Price) > 0 {
+			priceI = decimal.NewFromBigInt(new(big.Int).SetBytes(ret.Balances[j].Metadata.Price), 0)
 		}
-
-		if string(ret.Balances[j].Metadata.Price) != "" {
-			priceJ, err = decimal.NewFromString(string(ret.Balances[j].Metadata.Price))
-			if err != nil {
-				logger.WithError(err).Errorf("error parsing string price value, price: %s", ret.Balances[j].Metadata.Price)
-			}
+		if len(ret.Balances[j].Metadata.Price) > 0 {
+			priceJ = decimal.NewFromBigInt(new(big.Int).SetBytes(ret.Balances[j].Metadata.Price), 0)
 		}
 
 		mulI := decimal.NewFromFloat(float64(10)).Pow(decimal.NewFromBigInt(new(big.Int).SetBytes(ret.Balances[i].Metadata.Decimals), 0))
```

Comment:
For this 4 line block:
Do you need a precision great than 16?
If so you would need `DivRound`.